lightblue/src/test/java/org/esbtools/eventhandler/lightblue/BulkLightblueRequesterTest.java

```diff
@@ -105,6 +107,41 @@ public void shouldThrowNoSuchElementExceptionIfResponseNotFound() throws Executi
     }
 
     @Test
+    public void shouldThrowTimeoutExceptionIfResponseTimeExceedsLimit()
+            throws ExecutionException, InterruptedException, LightblueException, TimeoutException {
+
+        SlowDataLightblueClient slowClient = new SlowDataLightblueClient(client);
+        requester = new BulkLightblueRequester(slowClient);
+
+        insertUser("cooltester2000");
+
+        slowClient.pauseBeforeRequests();
+        DataFindRequest findTester = findUserByUsername("cooltester2000");
+
+        expectedException.expect(Matchers.instanceOf(TimeoutException.class));
+
+        requester.request(findTester).transformSync(responses -> {
+            return responses.forRequest(findTester).parseProcessed(TestUser.class);
+        }).get(50, TimeUnit.MILLISECONDS);
+
+    }
+    
+    @Test
+    public void shouldCompleteSuccessfullyIfResponseTimeDoesNotExceedLimit()
+            throws ExecutionException, InterruptedException, LightblueException, TimeoutException {
+
+        insertUser("cooltester2000");
+        DataFindRequest findTester = findUserByUsername("cooltester2000");
+
+        String userName = requester.request(findTester).transformSync(responses -> {
+            return responses.forRequest(findTester).parseProcessed(TestUser.class);
+        }).get(50, TimeUnit.MILLISECONDS).getUsername();
```

Comment:
This timeout should be kept longer ... we don't want to get false negatives on our build server because lightblue took 51 milliseconds to respond, for example. I would set it longer than the embedded lightblue should ever take, for example.